.github/workflows/ci.yml

```diff
@@ -179,31 +186,144 @@ jobs:
           echo "POLICY_TF_VAR_FILE=${VAR_FILE}" >> $GITHUB_ENV;
       - name: 'Init the terraform infrastructure for mapping'
         run: |-
-          cp -r ${{ env.TF_MODULE_DIR}} ${{ env.MAPPING_TF_MODULE_DIR}}
-          terraform -chdir=${{ env.MAPPING_TF_MODULE_DIR}} init
+          cp -r ${{ env.TF_MODULE_DIR }}/* ${{ runner.temp }}/${{ env.MAPPING_TF_MODULE_DIR }}
+          terraform -chdir=${{ runner.temp }}/${{ env.MAPPING_TF_MODULE_DIR }} init
       - name: 'Apply the infrastructure updates for mapping'
         id: 'tf_apply_mapping'
         run: |-
-          terraform -chdir=${{ env.MAPPING_TF_MODULE_DIR}} apply -auto-approve -var-file=${{env.MAPPING_TF_VAR_FILE}};
+          terraform -chdir=${{ runner.temp }}/${{ env.MAPPING_TF_MODULE_DIR }} apply -auto-approve -var-file=${{ env.MAPPING_TF_VAR_FILE }};
       - name: 'Init the terraform infrastructure for policy'
         run: |-
-          cp -r ${{ env.TF_MODULE_DIR}} ${{ env.POLICY_TF_MODULE_DIR}}
-          terraform -chdir=${{ env.POLICY_TF_MODULE_DIR}} init
+          cp -r ${{ env.TF_MODULE_DIR }}/* ${{ runner.temp }}/${{ env.POLICY_TF_MODULE_DIR }}
+          terraform -chdir=${{ runner.temp }}/${{ env.POLICY_TF_MODULE_DIR }} init
       - name: 'Apply the infrastructure updates for policy'
         id: 'tf_apply_policy'
         run: |-
-          terraform -chdir=${{ env.POLICY_TF_MODULE_DIR}} apply -auto-approve -var-file=${{env.POLICY_TF_VAR_FILE}};
+          terraform -chdir=${{ runner.temp }}/${{ env.POLICY_TF_MODULE_DIR }} apply -auto-approve -var-file=${{ env.POLICY_TF_VAR_FILE }};
+      - name: 'Upload terraform files for mapping'
+        uses: 'actions/upload-artifact@65d862660abb392b8c4a3d1195a2108db131dd05' # actions/upload-artifact@v3
+        if: '${{ steps.tf_apply_mapping.outcome == ''success'' }}'
+        with:
+          name: 'terraform-mapping-files-${{ github.run_id}}'
+          path: |
+            ${{ env.MAPPING_TF_VAR_FILE }}
+            ${{ runner.temp }}/${{ env.MAPPING_TF_MODULE_DIR }}/terraform.tfstate
+            ${{ runner.temp }}/${{ env.MAPPING_TF_MODULE_DIR }}/terraform.tfstate.backup
+
+          retention-days: 1
+      - name: 'Upload terraform files for policy'
+        uses: 'actions/upload-artifact@65d862660abb392b8c4a3d1195a2108db131dd05' # actions/upload-artifact@v3
+        if: '${{ steps.tf_apply_policy.outcome == ''success'' }}'
+        with:
+          name: 'terraform-policy-files-${{ github.run_id}}'
+          path: |
+            ${{ env.POLICY_TF_VAR_FILE }}
+            ${{ runner.temp }}/${{ env.POLICY_TF_MODULE_DIR }}/terraform.tfstate
+            ${{ runner.temp }}/${{ env.POLICY_TF_MODULE_DIR }}/terraform.tfstate.backup
+          retention-days: 1
+
+  # Call reusable workflow
+  call_mapping_reusable_workflow:
+    needs: ['set_up_integration_infra']
+    uses: 'abcxyz/pmap/.github/workflows/snapshot-file-copy.yml@main' # ratchet:exclude
```

Comment:
If you use main here, you can't catch the bugs exist in the PR that introduce bugs.  You need to use the version of the PR that runs the CI/CD workflow.

Reply:
Fixed